tests/pytests/unit/utils/test_gitfs_locks.py

```diff
@@ -0,0 +1,613 @@
+                rdict["sock_dir"],
+                conf_dir,
+            ],
+            self._user,
+            root_dir=rdict["root_dir"],
+        )
+
+        rdict["conf_file"] = pathlib.PurePath(conf_dir).joinpath(config_for)
+        with salt.utils.files.fopen(rdict["conf_file"], "w") as wfh:
+            salt.utils.yaml.safe_dump(rdict, wfh, default_flow_style=False)
+        return rdict
+
+    def get_config(
+        self,
+        config_for,
+        from_scratch=False,
+    ):
+        if from_scratch:
+            if config_for in ("master"):
+                return salt.config.master_config(self._master_cfg)
+            elif config_for in ("minion"):
+                return salt.config.minion_config(self._minion_cfg)
+            elif config_for == "client_config":
+                return salt.config_client_config(self._master_cfg)
+        if config_for not in ("master", "minion", "client_config"):
+            if config_for in ("master"):
+                return freeze(salt.config.master_config(self._master_cfg))
+            elif config_for in ("minion"):
+                return freeze(salt.config.minion_config(self._minion_cfg))
+            elif config_for == "client_config":
+                return freeze(salt.config.client_config(self._master_cfg))
+
+        log.error(
+            "Should not reach this section of code for get_config, missing support for input config_for %s",
+            config_for,
+        )
+
+        # at least return master's config
+        return freeze(salt.config.master_config(self._master_cfg))
+
+    @property
+    def config_dir(self):
+        return str(pathlib.PurePath(self._master_cfg).parent)
+
+    def get_config_dir(self):
+        log.warning("Use the config_dir attribute instead of calling get_config_dir()")
+        return self.config_dir
+
+    def get_config_file_path(self, filename):
+        if filename == "master":
+            return str(self._master_cfg)
+
+        if filename == "minion":
+            return str(self._minion_cfg)
+
+        return str(self._master_cfg)
+
+    @property
+    def master_opts(self):
+        """
+        Return the options used for the master
+        """
+        return self.get_config("master")
+
+    @property
+    def minion_opts(self):
+        """
+        Return the options used for the minion
+        """
+        return self.get_config("minion")
+
+
+class MyGitBase(AdaptedConfigurationTestCaseMixin):
+    """
+    mocked GitFS provider leveraging tmp_path
+    """
+
+    def __init__(
+        self,
+        salt_factories,
+        salt_master_factory,
+        salt_minion_factory,
+        tmp_path,
+    ):
+        super().__init__(
+            salt_factories,
+            salt_master_factory,
+            salt_minion_factory,
+            tmp_path,
+        )
+
+        tmp_name = self._tmp_name.join("/git_test")
+        pathlib.Path(tmp_name).mkdir(exist_ok=True, parents=True)
+
+        class MockedProvider(
```

Comment:
This is potentially the only valid class for this test module, and it should be defined at the global scope

Reply:
see comment above, and this was how the original test was working, and wanted to keep similar, rather than completely write a new test from scratch

Reply:
Refactored, take a look and see if that is sufficient, since given last Monday, want to move on and spend on other issues